OrbitCore/LinuxTracingHandler.cpp

```diff
@@ -43,41 +42,23 @@ void LinuxTracingHandler::OnSchedulingSlice(SchedulingSlice scheduling_slice) {
   tracing_buffer_->RecordTimer(std::move(timer));
 }
 
-void LinuxTracingHandler::OnCallstack(
-    const LinuxTracing::Callstack& callstack) {
+void LinuxTracingHandler::OnCallstackSample(CallstackSample callstack_sample) {
   CallStack cs;
-  cs.m_ThreadId = callstack.GetTid();
-
-  for (const auto& frame : callstack.GetFrames()) {
-    uint64_t address = frame.GetPc();
-    cs.m_Data.push_back(address);
-
-    // TODO(kuebler): This is mainly for clustering IPs to their functions.
-    //  We should enable this also as a post-processing step.
-    if (frame.GetFunctionOffset() !=
-        LinuxTracing::CallstackFrame::kUnknownFunctionOffset) {
-      absl::MutexLock lock{&addresses_seen_mutex_};
-      if (!addresses_seen_.contains(address)) {
-        LinuxAddressInfo address_info{address, frame.GetMapName(),
-                                      llvm::demangle(frame.GetFunctionName()),
-                                      frame.GetFunctionOffset()};
-        tracing_buffer_->RecordAddressInfo(std::move(address_info));
-        addresses_seen_.insert(address);
-      }
-    }
+  cs.m_ThreadId = callstack_sample.tid();
+  for (const auto& frame : callstack_sample.callstack().pcs()) {
```

Comment:
do you want to CHECK that you get the correct one_of here?

Reply:
Yes, done.